### PR TITLE
Add buck2 package.

### PR DESCRIPTION
**Pull request by dlorenc**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [X] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [X] REQUIRED - The version of the package is still receiving security updates
- [X] REQUIRED - The package is added to `packages.txt`

### CODE REVIEW

**Comment by dlorenc** [comment]
cc @kaniini for the help getting rustup working :)

File: buck2.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  commit: 54decd83a2adf998ec9f1a52958d726924241f32
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/facebook/buck2
+      # They don't have any releases yet, so use a commit below.
+      branch: main
+  - name: Configure and build
+    runs: |
+      git checkout ${{vars.commit}}
```
Comment (luhring) [inline buck2.yaml:27]: We should probably open an issue for this to be supported as a Melange pipeline or something

File: buck2.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      # This build requires a specific version of rust nightly.
+      # Be sure to check docs when bumping to more recent commits.
+      rustup install nightly-2023-01-24
+      export PATH="$HOME/.rustup/toolchains/nightly-2023-01-24-x86_64-unknown-linux-gnu/bin:$PATH"
```
Comment (kaniini) [inline buck2.yaml:31]:
```suggestion
      export PATH="$HOME/.rustup/toolchains/nightly-2023-01-24-${{host.triplet.rust}}/bin:$PATH"
```

File: buck2.yaml (hunk 1 above)
Comment (dlorenc) [reply buck2.yaml:27]: They changed it up a bit so there's now a `latest` tag that gets bumped semi-regularly. It makes our side easier but is still very unusual.

Comment (luhring) [reply buck2.yaml]: Interesting!

**Comment by dlorenc** [comment]
Marking this as ready for review. They've published a release, kind of.

**Review by imjasonh — APPROVED** [review]

**Pull request merged**